requirements-dev.txt

```diff
@@ -1,2 +1,2 @@
 pytest==8.3.3
-pre-commit
+pre-commit==3.8.0
```

Comment:
For some dependencies the exact versions don't matter and pinning them can even increase maintenance workload:
* Dependabot making PRs about every small version increment that doesn't actually matter
* Devs won't actually `pip install -r requirements-dev.txt` every time they continue working on the project

For the `requirements.txt` I would even consider it bad practise to pin all versions exactly, because it makes it very hard to install the library into an environment also containing other packages.

Fixing it for the Docker-build, however, would be good practice (as long as it's kept updated by dependabot).

Reply:
Thanks for the explanation.